File: Data Structures & Algorithms/4sum/submission-7.cpp
class Solution {
public:
    vector<vector<int>> fourSum(vector<int>& nums, int target) {
        set<vector<int>> res;
        sort(nums.begin(),nums.end());
        
        int n = nums.size();

        for (int i = 0; i <n;i++) {
            for (int j = i+1; j < n;j++) {
                for (int k = j+1; k < n; k++) {
                    for (int l = k+1; l<n; l++) {
                        if ((long long) nums[i] + nums[j] + nums[k] + nums[l] == target) {
                            res.insert({nums[i],nums[j],nums[k],nums[l]});
                        }
                    }
                }
            }
        }

        return vector<vector<int>>(res.begin(), res.end());
    }
};

File: Data Structures & Algorithms/sliding-window-maximum/submission-7.py
class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:
        l = 0
        r = k

        res = []

        while r <= len(nums):
            current = max(nums[l:r])
            res.append(current)
            l += 1
            r += 1

        return res